dracut/30ignition/ignition-setup.sh

```diff
@@ -1,22 +1,29 @@
 #!/bin/bash
+set -eu
 
-set -e
+copy_file() {
+    src="${1}"; dst="${2}"
+    if [ -f "${src}" ]; then
+        echo "Copying ${src} to ${dst}"
+        cp "${src}" "${dst}"
+    else
+        echo "File ${src} does not exist.. Skipping copy"
+    fi
+}
 
-# OEM directory in the initramfs itself
-src=/usr/share/oem
+destination=/usr/lib/ignition/
+mkdir -p $destination
 
-dst=/usr/lib/ignition
-mkdir -p "${dst}"
-for name in base.ign default.ign; do
-    if [[ -e "${src}/base/${name}" ]]; then
-        cp "${src}/base/${name}" "${dst}"
-    fi
-done
-if [[ -e "${src}/config.ign" ]]; then
-    cp "${src}/config.ign" "${dst}/user.ign"
-fi
+# We will support grabbing a platform specific base.ign config
+# from the initrd at /usr/lib/ignition/platform/${OEM_ID}/base.ign
+copy_file "/usr/lib/ignition/platform/${OEM_ID}/base.ign" "${destination}/base.ign"
 
-# if we have config.ign on boot, overwrite it
-if [[ -e "/boot/config.ign" ]]; then
-    cp "/boot/config.ign" "${dst}/user.ign"
-fi
+# We will support a user embedded config in the boot partition
+# under $bootmnt/ignition/config.ign
+bootmnt=/mnt/boot_partition
+mkdir -p $bootmnt
+mount /dev/disk/by-label/boot $bootmnt
+copy_file "${bootmnt}/ignition/config.ign" "${destination}/user.ign"
+# This script is run from a systemd unit with MountFlags=slave. This
+# unmount isn't strictly necessary, but we still exercise it here.
+umount $bootmnt
```

Comment:
But then the script may fail since we have `set -e`, right? Might as well remove it completely and rely on systemd.

Reply:
Agreed.

Reply:
+1